LDMP/calculate.py

```diff
@@ -714,6 +756,76 @@ def check_overwrites(self):
         return True
 
 
+class CalculationHidedOutputWidget(QtWidgets.QWidget, Ui_WidgetCalculationOutput):
+    def __init__(self, suffixes, subclass_name, parent=None):
+        super(CalculationHidedOutputWidget, self).__init__(parent)
+
+        self.output_suffixes = suffixes
+        self.subclass_name = subclass_name
+
+        self.process_id = None
+        self.process_datetime = None
+        self.process_datetime_str = None
+
+        self.setupUi(self)
+        self.hide()
+
+        # set default output basename
+        # commented and executed just before run to allow 
+        # setting new process_id
+        #self.set_output_basename() 
+
+    def set_output_basename(self):
+        """Set default ouptut filename of the local calculation.
+        """
+        self.process_id = str(uuid.uuid4())
+        self.process_datetime = datetime.now()
+        self.process_datetime_str = self.process_datetime.isoformat()
+
+        # TODO: check where used "LDMP/output_dir" to avoid side effects
+
+        # path where to store result
+        base_data_directory = QSettings().value("trends_earth/advanced/base_data_directory", None, type=str)
+        if not base_data_directory:
+            return
```

Comment:
shouldn't we use a default base data directory instead of returning here?

Reply:
`base_data_directory` is the default and have to be set otherwise nothing would work. I would set the default value in setting GUI

Reply:
> Yes... I face the issue, and I'm working on it.

hehehe... I wrongly removed the enum because was between two block of code that was marked as TODO: remove... I didnt 'expected some code was inserted in between :)

Reply:
> I would set the default value in setting GUI

@luipir does that mean the value will always be set?

Reply:
> @luipir does that mean the value will always be set?

right! it is the folder where all datasets and jobs are stored